### PR TITLE
Add ModelRotationEvent

### PR DESCRIPTION
**Pull request by Flemmli97**
Adds an event in RenderLivingBase.doRender just before the actual model is rendered. This allows to modify the models rotation etc. (including the players).

### CODE REVIEW

File: src/main/java/net/minecraftforge/client/event/ModelRotationEvent.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package net.minecraftforge.client.event;
+
+import net.minecraft.client.renderer.entity.RenderLivingBase;
+import net.minecraft.entity.Entity;
+import net.minecraftforge.fml.common.eventhandler.Event;
+
+/**
+ * Allows modification to the entity model before being rendered.
+ */
+public class ModelRotationEvent extends Event{
```
Comment (Landmaster) [inline src/main/java/net/minecraftforge/client/event/ModelRotationEvent.java]: Braces should be on their own line as per Forge convention.

**Review by Landmaster — CHANGES_REQUESTED** [review]
@Flemmli97 Can you add a simple test mod to make sure that the event works? Also:

**Comment by Flemmli97** [comment]
This now partially works. All of the models of vanilla minecraft are calling ModelBase.setRotationAngles in ModelBase.doRender again (for some reasons) and thus changes to the model made with this event wont get applied (i dont understand why they are calling that method twice). You can however disable the rendering of certain parts of the model which i did in the example mod (and then render the hidden parts yourself). Though this wasnt actually the purpose of this event.

**Review by Landmaster — CHANGES_REQUESTED** [review]
Please replace the tabs in `ModelRotationEvent` and `ModelRotationEventTest` with 4 spaces as per Forge convention.

**Comment by Darkhax** [comment]
What is the use of this PR? It seems like it would add a lot of overhead to rendering models.

**Comment by Jeryn99** [comment]
I think they indeed to allow the manipulation limbs for the use of custom animations to pre-existing mobs and players

**Comment by tterrag1098** [comment]
Again, this is a high-volume event with no justification. Please provide at least a use case for this event.

**Comment by Flemmli97** [comment]
the purpose was to modify the model before rendering it but since vanilla sets the rotations in ModelBase.doRender again it wont work anymore. i left it open thinking since it can work for modded entity models it might be useful but seems not like it. ill close this one

**Pull request closed without merging**